### PR TITLE
Add option for setting the clang importer

### PR DESCRIPTION
**Pull request by goldhoorn**

### CODE REVIEW

**Comment by goldhoorn** [comment]
@doudou: conceptional thoughts on this?

**Comment by doudou** [comment]
I don't like this non-solution to a real problem, but hey, what can I do about it ?

**Comment by goldhoorn** [comment]
It's at least one step in the right direction.
The user would get the info that clang is not yet (well) supported, AND (if in the future) it works. Then we can use it and only remove this message?!

**Comment by doudou** [comment]
> Then we can use it and only remove this message?!

**Comment by doudou** [comment]
> Then we can use it and only remove this message?!

Honestly, I don't want the clang importer to be the nominal importer for typelib. The amount of maintenance effort spent on Rock by anyone else than me is currently close to zero, so adding something as complex ? No way. castxml is definitely the only way forward.

**Comment by doudou** [comment]
>  The amount of maintenance effort spent on Rock by anyone else than me is currently close to zero

Let me rephrase this is a way that's a lot more fair to everyone involved.

**Since you and I are 99.99% of the amount of maintenance effort spent on Rock**, do you really want to add something as complex as the clang importer to it, while there is castxml that someone reliable (itk ...) will be maintaining for us.

**Comment by goldhoorn** [comment]
Don't know. I either want to do really support on this level of the toolchain at all. Neither for clang nor castxml.

However, the clang is (with https://github.com/orocos-toolchain/typelib/pull/72) alt least working.
The castxml seems farther away than the clang right now.
So without spending _yet_ much work to the solutions clang seems to be easier.

Other option would be who enable clang only on debian-testing and ubuntu15.10. 
So the question is: do we want to have a half-baked solution now, or do we wait until someone provides us a fully working solution?

I (currently) prefer a half-baked solution (with a warning that this is not fully supported), if errors occur and someone yields he can fix it.

**Comment by goldhoorn** [comment]
@2maz : i updated the PR in regards to our discussion yesterday
@doudou : I discussed this with 2maz, the outcome is that even this experimental clang is better than nothing on the newer OSes. We not officially support this but would make the testing easisier. Within the next weeks we will contact you with details how we here thing we would continue with typelib.

**Comment by 2maz** [comment]
While castxml will be an option, we have common issues with either solution. The clang importer is already in use and while it is experimental, is appears to be stable enough to allow for wider use, which allows to get some feedback. Any support seems better than no support on the named platforms.

Right now I would vote for merge so that we can nail down our roadmap in the meantime -- I put down some notes from a brief gathering:  https://rock.opendfki.de/wiki/WikiStart/OngoingWork/TypelibImporter

**Comment by doudou** [comment]
One major issue with the "roadmap" is the amount of effort that can be spent on it. Because so far it does not look good.

Other more technical aspects added at the bottom of the wiki document

**Comment by goldhoorn** [comment]
@doudou : and the thoughts on this PR (for the meantime) 2maz agreed to merge this but i don't know your current work. If you have no real time to get the castxml (in near time) running i wold like to merge this.?

**Comment by doudou** [comment]
No. I won't like to merge this. Now, if you guys are ready to spend ressources maintain it, then given the low maintenance effort on Rock anyways, just merge it. It won't make things incredibly worse than they already are.

**Comment by doudou** [comment]
Superseded #24 and https://github.com/rock-core/package_set/pull/95
